.github/workflows/run_tests.yml

```diff
@@ -1,62 +1,100 @@
 # This workflow will install Python dependencies, run tests and lint with a variety of Python versions
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
-name: Run tests
+name: Run tests on all platforms
 
 on:
   # Trigger the workflow on push but only for the master branch
   push:
     branches:
       - master
+  # Trigger the workflow on pull requests, but reveal no secrets, do not use pull_request_target, see:
+  # https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target
   pull_request:
 
+env:
+  # Even when given -y, apt will still sometimes hang at a prompt if a package
+  # has clarifications to ask; DEBIAN_FRONTEND=noninteractive prevents that,
+  # This will be defined for non-debian platforms below too, but there's no harm in that.
+  # (TravisCI quietly defined this on all their platforms, but we have to give it manually on GithubCI.)
+  DEBIAN_FRONTEND: 'noninteractive'
+  HDF5_USE_FILE_LOCKING: 'FALSE'
+
 jobs:
-  test:
+  ultra_matrix_test:
+    name: Matrix Test of Python ${{ matrix.python-version }} on ${{ matrix.os }}
 
-    runs-on: ubuntu-latest
+    # Matrix driven OS
+    runs-on: ${{ matrix.os }}
+
+    # Default shell for ALL subsequent steps.
+    defaults:
+      run:
+        shell: bash {0}
+
+    # Defining matrix for OS and Python
     strategy:
+
+      # Ensure exhaustive testing of all scenarios to gather maximum amount of information
+      fail-fast: false
+
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        os: [ "macos-11", "windows-latest", "ubuntu-18.04", "ubuntu-20.04",  "macos-10.15" ]
```

Comment:
We use "macos-11" here is because "macos-latest" is equals to "macos-10.15" instead of "macos-11".

Reply:
@dyt811 
macOS 11 is not available in GitHub actions yet. That is why the tests are failing. See [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners).

> Note: The macOS 11 virtual environment is currently provided as a private preview only. Any users or organizations that are already using this runner can continue using it, but we're not accepting any further users or organizations at this time. The macos-latest YAML workflow label still uses the macOS 10.15 virtual environment.

Reply:
Good point. I thought it was available but didn't read the fine print about it being "private preview only". Gonna revert that and merge PR then.